packages/firestore/test/unit/lite-api/types.test.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0x00 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0x00
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { expect } from 'chai';
+
+import {
+  DocumentData,
+  DocumentReference,
+  WithFieldValue,
+  PartialWithFieldValue,
+  SetOptions
+} from '../../../src/lite-api/reference';
+import {
+  getDoc,
+  setDoc,
+  updateDoc
+} from '../../../src/lite-api/reference_impl';
+import {
+  FirestoreDataConverter,
+  QueryDocumentSnapshot
+} from '../../../src/lite-api/snapshot';
+
+describe('FirestoreTypeConverter', () => {
```

Comment:
Great tests. The only thing I would consider adding is some negative test cases.

Like this test case that would prevent relaxing of the type in the future.
```
    interface MyModelType {
      stringProperty: string;
      numberProperty: number;
    }
    interface MyDbType {
      a: string;
      b: number;
    }
    const converter: FirestoreDataConverter<MyModelType, MyDbType> = {
     // @ts-expect-error
      toFirestore(obj: WithFieldValue<MyDbType>) {
        return { a: obj.stringProperty, b: obj.numberProperty };
      },
     // @ts-expect-error
      fromFirestore(snapshot: QueryDocumentSnapshot<MyAppType>) {
        return {
          stringProperty: snapshot.data().a,
          numberProperty: snapshot.data().b
        };
      }
```

Or even something simple like:
```
// @ts-expect-error
await setDoc(newDocRef, { doesThisInterfaceActuallyEnforceTypes: 'foo', numberProperty: 'not a number' });
````

Reply:
Sure, I can add some negative test cases like this. Good idea.

Reply:
Done in https://github.com/firebase/firebase-js-sdk/pull/7932